sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -88,7 +88,10 @@ class StringType private(val collationId: Int) extends AtomicType with Serializa
  */
 @Stable
 case object StringType extends StringType(0) {
-  private[spark] def apply(collationId: Int): StringType = new StringType(collationId)
+  private[spark] def apply(collationId: Int): StringType = {
```

Comment:
I checked several callers of this function. The input collation id is mostly calculated from collation name. This assertion doesn't seem to be necessary and it's not cheap. Shall we revert?

I'm fine with other cleanups in this PR.

Reply:
Okay, I have revert it. 
Thank you!